Octree/include/CGAL/Octree/Node.h

```diff
@@ -0,0 +1,346 @@
+   * \param index The index of the child node, as an int
+   * \return A reference to the node
+   */
+  Node<Point_index> &operator[](int index) {
+
+    assert(!is_leaf());
+    assert(0 <= index && index < 8);
+
+    return (*m_children)[index];
+  }
+
+  /*!
+   * \brief Read-only access the child nodes of this node by their indices
+   *
+   * \param index The index of the child node, as an int
+   * \return A const reference to the node
+   */
+  const Node<Point_index> &operator[](int index) const {
+
+    assert(!is_leaf());
+    assert(0 <= index && index < 8);
+
+    return (*m_children)[index];
+  }
+
+  /// @}
+
+  /// \name Property Accessors
+  /// @{
+
+  /*!
+   * \brief Read-only access to this node's parent
+   *
+   * Ownership of a node is not equivalent to ownership of the entire tree,
+   * so it's not possible to obtain write access to a node's parent,
+   * only its children.
+   * Note that the return type is nullable,
+   * attempting to find the parent of a root node will return null.
+   * \todo Should I instead assert the node isn't root? (that would make this undefined behavior)
+   *
+   * \return A const pointer to the parent of this node (possibly nullptr)
+   */
+  const Node<Point_index> *parent() const { return m_parent; }
+
+  /*!
+   * \brief Retrieve this node's depth in the tree
+   * \return the depth of this node, where root has a depth of 0
+   */
+  const uint8_t &depth() const { return m_depth; }
+
+  /*!
+   * \brief Retrieve this node's location in the tree
+   *
+   * \todo Should I link to an explanation of the location type?
+   *
+   * \return this node's location
+   */
+  Int_location location() const {
+
+    return m_location;
+  }
+
+  /*!
+   * \brief Retrieve this node's index in relation to its parent
+   * \return the index of this nod3
+   */
+  Index index() const {
+
+    // TODO: There must be a better way of doing this!
+
+    Index result;
+
+    result[0] = location()[0] & 1;
+    result[1] = location()[1] & 1;
+    result[2] = location()[2] & 1;
+
+    return result;
+  }
+
+  /*!
+   * \brief Determine whether this node is a leaf node
+   * \return whether this node has no children
+   */
+  bool is_leaf() const { return (!m_children); }
+
+  /*!
+   * \brief Determine whether this node is the root node
+   * \return whether this node has no parent
+   */
+  bool is_root() const { return (!m_parent); }
+
+  /// @}
+
+  /// \name Value Accessors
+  /// @{
+
+  /*!
+   * \brief Access to the content held by this node
+   * \return a reference to the collection of point indices
+   */
+  Point_range &points() { return m_points; }
+
+  /*!
+   * \brief Read-only access to the content held by this node
+   * \return a read-only reference to the collection of point indices
+   */
+  const Point_range &points() const { return m_points; }
+
+  /*!
+   * \brief Check whether this node contains any points
+   * \return if this node contains no points
+   */
+  bool is_empty() const {
```

Comment:
`empty()`, that is no is_  prefix

Reply:
Good point, that definitely makes it more consistent with other parts of CGAL. Does it make sense to change `is_leaf` and `is_root` too?

Reply:
If you search for `is_` in CGAL, you will find many occurrences, so I think you can keep `is_leaf()` and `is_root()`.

Don't use `is_` if the function obviously returns a boolean (as is the case for `empty()`, which by the way is also more consistent with STL containers).

Functions `leaf()` or `root()` may sound like they would return _the_ leaf object or _the_ root object, and not a boolean, so keeping `is_` makes sense there.

Reply:
"is_"  is fine.

Reply:
That makes a lot of sense! In that case, I think this can be resolved.